graphql/resolvers.go

```diff
@@ -365,7 +366,30 @@ func (r *mutationResolver) SpawnHost(ctx context.Context, spawnHostInput *SpawnH
 		options.Expiration = spawnHostInput.Expiration
 	}
 
+	if util.IsPtrSetToTrue(spawnHostInput.UseProjectSetupScript) {
+		if spawnHostInput.TaskID == nil {
+			return nil, ResourceNotFound.Send(ctx, "A valid task id must be supplied when useProjectSetupScript is set to true")
+		}
+		options.UseProjectSetupScript = *spawnHostInput.UseProjectSetupScript
+		options.TaskID = *spawnHostInput.TaskID
+	}
+
 	hc := &data.DBConnector{}
+
+	if util.IsPtrSetToTrue(spawnHostInput.SpawnHostsStartedByTask) {
+		if spawnHostInput.TaskID == nil {
+			return nil, ResourceNotFound.Send(ctx, "A valid task id must be supplied when SpawnHostsStartedByTask is set to true")
+		}
+		task, e := task.FindOneId(*spawnHostInput.TaskID)
```

Comment:
Should this be set to err? otherwise you're checking something unrelated on the next line (same with 387)